src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingPassBase.cs

```diff
@@ -581,16 +578,16 @@ bool IsImplicitStatementStart()
                 // `@code {|...` - false
                 //
 
-                var previousCharIndex = absoluteIndex - 1;
-                var previousChar = sourceText[previousCharIndex];
-                if (previousChar != '@')
+                if (owner.SpanStart == mappingSpan.Start &&
+                    owner is CSharpStatementLiteralSyntax &&
+                    owner.Parent is CSharpCodeBlockSyntax &&
+                    owner.PreviousSpan() is CSharpTransitionSyntax)
```

Comment:
Didn't add a new comment for this because I think the one above already captures what this does.